src/hooks/useDrafts/useDrafts.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import { addDraft, clearDrafts, getDraft, getDrafts, removeDraft, updateDraft } from './utils'
+
+export type CommonDraftProps = {
+  id: string
+  updatedAt?: string
+}
+
+export type DraftType = 'videos' | 'channels'
+
+export type VideoDraft = {
+  type: 'videos'
+  title?: string
+  description?: string
+  publicness?: 'public' | 'private'
+  publishedBefore?: string
+  marketing?: boolean
+  contentRating?: 'all' | 'mature'
+} & CommonDraftProps
+
+export type ChannelDraft = {
```

Comment:
I'm not sure if we should include this here. We will not be using channel drafts for this release. General drafts handling is fine, but this will be dead, unused code for the foreseeable future so I think we should just add `VideoDraft` now